sdks/java/io/iceberg/src/main/java/org/apache/beam/sdk/io/iceberg/FilterUtils.java

```diff
@@ -0,0 +1,283 @@
+class FilterUtils {
+  private static final Map<SqlKind, Operation> FILTERS =
+      ImmutableMap.<SqlKind, Operation>builder()
+          .put(SqlKind.IS_NULL, Operation.IS_NULL)
+          .put(SqlKind.IS_NOT_NULL, Operation.NOT_NULL)
+          .put(SqlKind.LESS_THAN, Operation.LT)
+          .put(SqlKind.LESS_THAN_OR_EQUAL, Operation.LT_EQ)
+          .put(SqlKind.GREATER_THAN, Operation.GT)
+          .put(SqlKind.GREATER_THAN_OR_EQUAL, Operation.GT_EQ)
+          .put(SqlKind.EQUALS, Operation.EQ)
+          .put(SqlKind.NOT_EQUALS, Operation.NOT_EQ)
+          .put(SqlKind.IN, Operation.IN)
+          .put(SqlKind.NOT_IN, Operation.NOT_IN)
+          .put(SqlKind.AND, Operation.AND)
+          .put(SqlKind.OR, Operation.OR)
+          .build();
+
+  static Expression convert(@Nullable String filter, Schema schema) {
+    if (filter == null) {
+      return Expressions.alwaysTrue();
+    }
+
+    SqlParser parser = SqlParser.create(filter);
+    try {
+      SqlNode expression = parser.parseExpression();
+      return convert(expression, schema);
+    } catch (Exception exception) {
+      throw new RuntimeException(
+          String.format("Encountered an error when parsing filter: '%s'", filter), exception);
+    }
+  }
+
+  private static Expression convert(SqlNode expression, Schema schema) throws SqlParseException {
+    checkArgument(expression instanceof SqlBasicCall);
+    SqlBasicCall call = (SqlBasicCall) expression;
+
+    SqlOperator op = call.getOperator();
+    SqlKind kind = op.getKind();
+
+    Operation operation =
+        checkArgumentNotNull(
+            FILTERS.get(kind),
+            "Unable to convert SQL operation '%s' in Iceberg expression: %s",
+            kind,
+            expression.toString());
+
+    switch (operation) {
+      case IS_NULL:
+        return Expressions.isNull(getOnlyChildName(call));
+      case NOT_NULL:
+        return Expressions.notNull(getOnlyChildName(call));
+      case LT:
+        return convertFieldAndLiteral(
+            Expressions::lessThan, Expressions::greaterThan, call, schema);
+      case LT_EQ:
+        return convertFieldAndLiteral(
+            Expressions::lessThanOrEqual, Expressions::greaterThanOrEqual, call, schema);
+      case GT:
+        return convertFieldAndLiteral(
+            Expressions::greaterThan, Expressions::lessThan, call, schema);
+      case GT_EQ:
+        return convertFieldAndLiteral(
+            Expressions::greaterThanOrEqual, Expressions::lessThanOrEqual, call, schema);
+      case EQ:
+        return convertFieldAndLiteral(
+            (ref, lit) -> {
+              if (lit == null) {
+                return Expressions.isNull(ref);
+              } else if (NaNUtil.isNaN(lit)) {
+                return Expressions.isNaN(ref);
+              } else {
+                return Expressions.equal(ref, lit);
+              }
+            },
+            call,
+            schema);
+      case NOT_EQ:
+        return convertFieldAndLiteral(
```

Comment:
Let's try to do 100% test coverage for this file.

Reply:
Tried to do that in FilterUtilsTest. Let me know if anything is missing